### PR TITLE
Convert `org.pkl.core.util` POJOs to `record` classes

### PR DESCRIPTION
**Pull request by KushalP**

### CODE REVIEW

File: pkl-core/src/main/java/org/pkl/core/util/GlobResolver.java [hunk 1]
```diff
@@ -45,26 +45,24 @@ public InvalidGlobPatternException(String message) {
...
 
+    @Deprecated
     public URI getUri() {
       return uri;
     }
 
+    @Deprecated
     public boolean isDirectory() {
       return isDirectory;
     }
```
Comment (bioball) [inline pkl-core/src/main/java/org/pkl/core/util/GlobResolver.java:68]: The `util` package is not a public API, so no need to add deprecated methods.

Comment (bioball) [inline pkl-core/src/main/java/org/pkl/core/util/GlobResolver.java:68]: The util package is not a public API, so no need to add deprecated methods.

```suggestion
  public record ResolvedGlobElement(String path, URI uri, boolean isDirectory) {
  }
```

**Review by bioball — CHANGES_REQUESTED** [review]
Thanks! Just one comment.

File: pkl-core/src/main/java/org/pkl/core/util/GlobResolver.java (hunk 1 above)
Comment (bioball) [reply pkl-core/src/main/java/org/pkl/core/util/GlobResolver.java]: Oh, we definitely want that, my bad!

File: pkl-core/src/main/java/org/pkl/core/util/yaml/snake/YamlResolver.java [hunk 2]
```diff
@@ -48,15 +48,7 @@ public Tag resolve(String value, Boolean implicit) {
...
-
-    ResolverTuple(Tag tag, Pattern regexp) {
-      this.tag = tag;
-      this.regexp = regexp;
-    }
-
+  private record ResolverTuple(Tag tag, Pattern regexp) {
     @Override
     public String toString() {
       return "Tuple tag=" + tag + " regexp=" + regexp;
```
Comment (KushalP) [inline pkl-core/src/main/java/org/pkl/core/util/yaml/snake/YamlResolver.java:54]: Some inconsistency here with the `toString()` format. Might be worth deleting this and using the one that comes with `record` ?

Comment (bioball) [reply pkl-core/src/main/java/org/pkl/core/util/yaml/snake/YamlResolver.java:54]: The `org.pkl.core.util.yaml.snake` package contains source code that is vendored from the SnakeYAML library.

**Review by bioball — APPROVED** [review]
LGTM

**Pull request merged**